lib/matplotlib/axis.py

```diff
@@ -2013,6 +2013,8 @@ def set_ticks_position(self, position):
         ----------
         position : {'top', 'bottom', 'both', 'default', 'none'}
         """
+        cbook._check_in_list(['top', 'bottom', 'both', 'default', 'none'],
+                             position=position)
```

Comment:
Little confused by this.  Do you need the `ValueError` anymore?

Reply:
Do you mean the `else` branch? - Not strictly, but I'd still keep it; see https://github.com/matplotlib/matplotlib/pull/15939#discussion_r357978387

Reply:
Then I’m not sure why we would bother w the cbook call.

Reply:
I'm not a fan of having both.

Reply:
agreed (didn't insist on removing it, but prefer without it)

Reply:
As written above
- `_check_in_list` gives the better error message.
- the else branch makes that claimed/checked values and actual code behavior do not get out of sync.

I think having an else clause in this particular case is justifiable. Otherwise no action would be silently performed if claimed/checked values and actual code behavior do not get out of sync. Yes, it's a bit defensive to guard against future code changes, but why take a risk?

I'd be fine with changing to a `RuntimeError` or some other exception, but getting to else is an error and should not be passed silently.

Reply:
I guess you can make the last line eg assert False.